SSH banner check without empty and unused user and password causes panic

### Describe the bug

```
endpoints:
  - name: Hetzner StorageBox SSH
    group: Infrastructure
    url: ssh://u497426.your-storagebox.de:23
    interval: 10s
    conditions:
      - "[CONNECTED] == true"
```

### What do you see?

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0xb0d722]

goroutine 373 [running]:
github.com/TwiN/gatus/v5/config/endpoint.(*Endpoint).call(0xc000241e60, 0xc000000620)
        /app/config/endpoint/endpoint.go:501 +0xae2
github.com/TwiN/gatus/v5/config/endpoint.(*Endpoint).EvaluateHealthWithContext(0xc000241e60, 0x0)
        /app/config/endpoint/endpoint.go:326 +0x3b9
github.com/TwiN/gatus/v5/config/endpoint.(*Endpoint).EvaluateHealth(...)
        /app/config/endpoint/endpoint.go:286
github.com/TwiN/gatus/v5/watchdog.executeEndpoint(0xc000241e60, 0xc000000000, {0x240f6a0, 0x0, 0x0})
        /app/watchdog/endpoint.go:49 +0x1f8
github.com/TwiN/gatus/v5/watchdog.monitorEndpoint(0xc000241e60, 0xc000000000, {0x240f6a0, 0x0, 0x0}, {0x186beb0, 0xc0000a9450})
        /app/watchdog/endpoint.go:17 +0x66
created by github.com/TwiN/gatus/v5/watchdog.Monitor in goroutine 1
        /app/watchdog/watchdog.go:42 +0x210
```

### What do you expect to see?

Gatus not crashing.

### List the steps that must be taken to reproduce this issue

1. Add check like above

### Version

5.27.1

### Additional information

_No response_

--- a/config/endpoint/endpoint.go
+++ b/config/endpoint/endpoint.go
@@ -498,7 +498,7 @@ func (e *Endpoint) call(result *Result) {
 		result.Duration = time.Since(startTime)
 	} else if endpointType == TypeSSH {
 		// If there's no username/password specified, attempt to validate just the SSH banner
-		if len(e.SSHConfig.Username) == 0 && len(e.SSHConfig.Password) == 0 {
+		if e.SSHConfig == nil || (len(e.SSHConfig.Username) == 0 && len(e.SSHConfig.Password) == 0) {
 			result.Connected, result.HTTPStatus, err = client.CheckSSHBanner(strings.TrimPrefix(e.URL, "ssh://"), e.ClientConfig)
 			if err != nil {
 				result.AddError(err.Error())

